### DISCUSSION TITLE
Config file location

### DISCUSSION THREAD

**Discussion by fdncred** [nushell/nushell#8263] [Polls]
We frequently get questions about why we use [dirs-next](https://crates.io/crates/dirs-next) to control where the configuration files for nushell get placed and read from.

**Comment by rgwood** [comment]
I'm thinking this was missing a 4th option:

- Wherever the XDG_CONFIG_HOME environment variable says it should go. If absent, default to `~/.config`

In retrospect I would have voted for that.

**Reply by rgwood** [reply]
To clarify, I think there are 4 possible approaches here:
|   | Ignore XDG_CONFIG_HOME if set | Respect XDG_CONFIG_HOME if set |
|---|---|---|
| **Use OS-specific directories** | 1  | 2  |
|  **Use `~/.config` on all OSs** |  3 | 4  |

4 wasn't an option on the poll.

**Comment by ammkrn** [comment]
I was surprised it wasn't in ~/.config on macos, but I can't speak to what the Windows default should be.

**Reply by rgwood** [reply]
My understanding is that `AppData\*` is the traditional location on Windows but these days it's common for tools to use `~\.config` for consistency across platforms.

My `~\.config` folder on Windows contains config for [Scoop](https://scoop.sh/), [Lazygit](https://github.com/jesseduffield/lazygit), [Starship](https://starship.rs/), and [Micro](https://micro-editor.github.io/).

**Comment by StevenDoesStuffs** [comment]
I have no clue how it should be on other platforms, but on Linux `XDG_CONFIG_HOME` really ought to be respected with a fallback to `~/.config`.

**Reply by ChrisDenton** [reply]
I'm confused, is this not what [dirs-next](https://crates.io/crates/dirs-next) already does?

> |**Function name**|**Value on Linux**|
> |--|--|
> |`config_dir`|`Some($XDG_CONFIG_HOME)` or `Some($HOME/.config)`|

**Reply by fdncred** [reply]
> I'm confused, is this not what [dirs-next](https://crates.io/crates/dirs-next) already does?
> 
> 
> 
> > |**Function name**|**Value on Linux**|
> 
> > |--|--|
> 
> > |`config_dir`|`Some($XDG_CONFIG_HOME)` or `Some($HOME/.config)`|

I don't think it does that all platforms.

**Reply by ChrisDenton** [reply]
Right but I was specifically commenting on the Linux situation. It is indeed different on Mac and Windows.

**Reply by fdncred** [reply]
Sorry, was on my phone and it's not as intuitive about what you were replying to. My mistake.
